arangod/Replication/ReplicationFeature.cpp

```diff
@@ -184,6 +185,16 @@ void ReplicationFeature::collectOptions(
           "Whether to use the newer revision-based replication protocol.",
           new BooleanParameter(&_syncByRevision))
       .setIntroducedIn(30700);
+
+  options
+      ->addOption("--replication.auto-repair-revision-trees",
+                  "Whether to automatically repair revision trees of shards "
+                  "after too many shard synchronization failures.",
+                  new BooleanParameter(&_autoRepairRevisionTrees),
+                  arangodb::options::makeFlags(
+                      arangodb::options::Flags::DefaultNoComponents,
+                      arangodb::options::Flags::OnDBServer))
+      .setIntroducedIn(31100);
```

Comment:
If we backport this to 3.10, we probably want to put the 3.10 version in which it appears here.

Reply:
Done